python_modules/dagster/dagster/core/execution/plan/inputs.py

```diff
@@ -394,6 +394,65 @@ def compute_version(
         return dagster_type.loader.compute_loaded_input_version(config_data)
 
 
+@whitelist_for_serdes
+class FromRootInputConfig(
```

Comment:
Why aren't we able to use `FromConfig`? And, if we can't, are we able to share a more of the implementation?

Reply:
I don't think we can share much here. The place we pull the input values from is totally different. Only method shared is required_resource_keys, and that's repeated for every single input source.